### PR TITLE
Support PyQt 5.6

### LINKED ISSUE #3073: Spyder doesn't work with QtWebEngine
## Description of your problem
I tried to launch spyder 3.0.0 beta2 from command line with qt5 installed from homebrew, sip and pyqt5 install from source code and spyder preview installed from pip, when I launched it, the error log said

Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/spyderlib/spyder.py", line 3119, in main
    mainwindow = run_spyder(app, options, args)
  File "/usr/local/lib/python3.5/site-packages/spyderlib/spyder.py", line 3005, in run_spyder
    main.setup()
  File "/usr/local/lib/python3.5/site-packages/spyderlib/spyder.py", line 822, in setup
    message=_("Spyder Internal Console\n\n"
  File "/usr/local/lib/python3.5/site-packages/spyderlib/plugins/console.py", line 79, in __init__
    self.find_widget.set_editor(self.shell)
  File "/usr/local/lib/python3.5/site-packages/spyderlib/widgets/findreplace.py", line 250, in set_editor
    from spyderlib.qt.QtWebKit import QWebView
  File "/usr/local/lib/python3.5/site-packages/spyderlib/qt/QtWebKit.py", line 10, in <module>
    from PyQt5.QtWebKitWidgets import QWebPage, QWebView      # analysis:ignore
ImportError: dlopen(/usr/local/lib/python3.5/site-packages/PyQt5/QtWebKitWidgets.so, 2): Library not loaded: /usr/local/opt/qt5/lib/QtWebKitWidgets.framework/Versions/5/QtWebKitWidgets
  Referenced from: /usr/local/lib/python3.5/site-packages/PyQt5/QtWebKitWidgets.so
  Reason: image not found

* Spyder Version: 3.0.0 beta2
* Python Version: 3.5.1
* Operating system: OS X 10.11 El Cap

### LINKED ISSUE #3145: Spyder doesn't work with Qt 5.6
## Description of your problem

**What steps will reproduce the problem?**

1. pip install PyQt5.6 on Windows 7 64 / Python3.5 32 bit
2. instal spyder master from github
3. launch spyder

**What is the expected output? What do you see instead?**
spyder doesn't start well (but i see the initial splash screen).
Looking for a log of error but I don't find it

**Please provide any additional information below**
Did someone tried  this already ? Is it supposed to work ?

## Versions and main components

* Spyder Version:spyder-3.0.0.dev0 of April 29th
* Python Version: 3.5.1 32 bit 
* Operating system: Windows 7 64

## Dependencies

Please go to the menu entry `Help > Optional Dependencies` (or
`Help > Dependencies`), press the button `Copy to clipboard`
and paste the contents below:

### PR DESCRIPTION
**Pull request by ccordoba12**
Fixes #3073
Fixes #3145

----

TODO:

- [x] Review all Qt slots to see if they have the right signatures (PyQt 5.6 fails with the wrong signatures).
- [x] Fix loading js and css files from local computer (WebEngine refuses to do it).

Pinging @stonebig and @Nodd about this one.

### CODE REVIEW

**Comment by Nodd** [comment]
I just launched spyder successfully. I'll let you know if there are any problems when using it.

File: spyderlib/widgets/browser.py [hunk 1]
```diff
@@ -10,10 +10,12 @@
 import sys
 
 # Third party imports
+from qtpy import PYQT5
```
Comment (Nodd) [inline spyderlib/widgets/browser.py]: I think that this constant is not used in the file.

Comment (ccordoba12) [inline spyderlib/widgets/browser.py]: You're right. I was using it before I noticed we need a more fine grained constant (`WEBENGINE`) to activate/deactivate some things according to WebEngine or WebKit.

**Comment by stonebig** [comment]
a picture is worth a thousand words:
spyder_qt56_windows

**Comment by ccordoba12** [comment]
Ok, this is ready! I'll merge it after I fix our Travis and AppVeyor integrations :-)

**Comment by stonebig** [comment]
Testing further, the help onobject is not working well, the "top_right' help aera doesn't work:
- I have no welcome message over it explaining to do "Ctrl+I"
- trying to type "numpy" on the object, I loss focus after each character is typed.
- crtl+I on the editor feed that obhect field but nothiong happen.

**Comment by ccordoba12** [comment]
@stonebig, I think the problem you're seeing is caused because the PyQt5 wheel is packaged incorrectly (at least that's what I discovered on Linux). WebEngine (the new and shiny Qt web backend) fails because it needs some files that are placed in another directory of the wheel.

However, I need to merge this one to keep testing things, so please open a new issue about your problem and I'll try to look at it when I have some time :-)

**Comment by stonebig** [comment]
Hi @ccordoba12 ,

Could you be specific about "it needs files that are placed in another directory of the wheel" ?

The PyQt maintainer may fix it, if enough details are provided

https://www.riverbankcomputing.com/pipermail/pyqt/2016-June/037636.html

**Comment by ccordoba12** [comment]
I'll check it out in a couple of hours on Windows.

On Linux I needed to move all files in

    site-packages/PyQt5/Qt/resources

to

    site-packages/PyQt5/Qt/libexec

so that they are placed next to the `QWebEngineProcess` executable.

Maybe you need to do the same on Windows :-)

**Comment by stonebig** [comment]
can you give me the list of files in this `site-packages/PyQt5/Qt/resources` ? I don't have the same directory but I should have the same detailed files.

**Comment by stonebig** [comment]
I copy:
- what is in `python-3.5.1\Lib\site-packages\PyQt5\Qt\resources\`
- in `python-3.5.1\Lib\site-packages\PyQt5\`

````
icudtl.dat
qtwebengine_resources.pak
qtwebengine_resources_100p.pak
qtwebengine_resources_200p.pak
````

then:
- First spyder start (loooong) then nothing, have to kill "python*." and "spyder.exe" process
- second "spyder reset"
- third "spyder" ...... and then I get locked out spyder, but the first "web" window did display.

**Comment by ccordoba12** [comment]
Those are the exact files I copied next to `QtWebEngineProcess`. Could you post a screenshot of what you have now?

**Comment by stonebig** [comment]
spyder hangs in this state

ptipython_ipython5b3_copy_paste

with these processes:
ptipython_ipython5b3_copy_paste2

**Comment by fepegar** [comment]
I have this problem too (should I open a new issue?):

I'm running:
```
➜  ~ conda create -n testenv python=3 spyder
➜  ~ . activate testenv
(testenv) ➜  ~ spyder
Traceback (most recent call last):
  File "/home/fernando/anaconda2/envs/testenv/lib/python3.6/site-packages/qtpy/QtWebEngineWidgets.py", line 22, in <module>
    from PyQt5.QtWebEngineWidgets import QWebEnginePage
ModuleNotFoundError: No module named 'PyQt5.QtWebEngineWidgets'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/fernando/anaconda2/envs/testenv/bin/spyder", line 6, in <module>
    sys.exit(spyder.app.start.main())
  File "/home/fernando/anaconda2/envs/testenv/lib/python3.6/site-packages/spyder/app/start.py", line 103, in main
    from spyder.app import mainwindow
  File "/home/fernando/anaconda2/envs/testenv/lib/python3.6/site-packages/spyder/app/mainwindow.py", line 92, in <module>
    from qtpy import QtWebEngineWidgets  # analysis:ignore
  File "/home/fernando/anaconda2/envs/testenv/lib/python3.6/site-packages/qtpy/QtWebEngineWidgets.py", line 26, in <module>
    from PyQt5.QtWebKitWidgets import QWebPage as QWebEnginePage
ModuleNotFoundError: No module named 'PyQt5.QtWebKitWidgets'
```

My system:
```
➜  ~ uname -a
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.2 LTS
Release:	16.04
Codename:	xenial
```

I can't find the directory you mentioned:
```
➜  ~ ls anaconda2/envs/testenv/lib/python3.6/site-packages/PyQt5/
__init__.py                     QtBluetooth.pyi  QtGui.pyi                QtNetwork.pyi       Qt.pyi              QtSql.pyi         QtWebSockets.pyi   QtXml.pyi
__pycache__                     QtBluetooth.so   QtGui.so                 QtNetwork.so        QtQml.pyi           QtSql.so          QtWebSockets.so    QtXml.so
_QOpenGLFunctions_2_0.pyi       QtCore.pyi       QtHelp.pyi               QtNfc.pyi           QtQml.so            QtSvg.pyi         QtWidgets.pyi      uic
_QOpenGLFunctions_2_0.so        QtCore.so        QtHelp.so                QtNfc.so            QtQuick.pyi         QtSvg.so          QtWidgets.so
_QOpenGLFunctions_2_1.pyi       QtDBus.pyi       QtMultimedia.pyi         QtOpenGL.pyi        QtQuick.so          QtTest.pyi        QtX11Extras.pyi
_QOpenGLFunctions_2_1.so        QtDBus.so        QtMultimedia.so          QtOpenGL.so         QtQuickWidgets.pyi  QtTest.so         QtX11Extras.so
_QOpenGLFunctions_4_1_Core.pyi  QtDesigner.pyi   QtMultimediaWidgets.pyi  QtPrintSupport.pyi  QtQuickWidgets.so   QtWebChannel.pyi  QtXmlPatterns.pyi
_QOpenGLFunctions_4_1_Core.so   QtDesigner.so    QtMultimediaWidgets.so   QtPrintSupport.so   Qt.so               QtWebChannel.so   QtXmlPatterns.so
```

**Pull request merged**